src/mcp-bundle/src/DependencyInjection/ContainerBuilder/McpToolChainCompilerPass.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\McpBundle\DependencyInjection\ContainerBuilder;
+
+use Symfony\AI\McpSdk\Capability\Tool\CollectionInterface;
+use Symfony\AI\McpSdk\Capability\Tool\ToolExecutorInterface;
+use Symfony\AI\McpSdk\Capability\ToolChain;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * @author Tom Hart <tom.hart.221@gmail.com>
+ */
+final class McpToolChainCompilerPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    public function process(ContainerBuilder $container): void
+    {
+        // If the user has already defined this service, don't override it.
+        if ($container->hasDefinition(ToolExecutorInterface::class)) {
+            return;
+        }
+
+        $definition = new Definition(ToolChain::class);
```

Comment:
you could always define a `mcp.tool_chain` service using a tagged iterator, and then defining a `ToolExecutorInterface` alias pointing at it (which projects can still override if they don't want to rely on the tool chain).
There is no need for this compiler pass IMO.